RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotFromDB.cc

```diff
@@ -0,0 +1,66 @@
+/**_________________________________________________________________
+   class:   OnlineBeamSpotFromDB.cc
+   package: RecoVertex/BeamSpotProducer
+   
+
+
+ author: Francisco Yumiceva, Fermilab (yumiceva@fnal.gov)
+
+
+________________________________________________________________**/
+
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/IOVSyncValue.h"
+#include "CondFormats/DataRecord/interface/BeamSpotTransientObjectsRcd.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotOnlineObjects.h"
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include <string>
+
+class OnlineBeamSpotFromDB : public edm::one::EDAnalyzer<> {
+public:
+  explicit OnlineBeamSpotFromDB(const edm::ParameterSet& iConfig);
+  ~OnlineBeamSpotFromDB() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& desc);
+  edm::ESGetToken<BeamSpotObjects, BeamSpotTransientObjectsRcd> bsToken_;
+
+private:
+  void analyze(const edm::Event&, const edm::EventSetup&) override;
+};
+
+OnlineBeamSpotFromDB::OnlineBeamSpotFromDB(const edm::ParameterSet& iConfig)
+    : bsToken_(esConsumes<BeamSpotObjects, BeamSpotTransientObjectsRcd>()) {}
+
+OnlineBeamSpotFromDB::~OnlineBeamSpotFromDB() {}
+
+void OnlineBeamSpotFromDB::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  //edm::ESHandle<BeamSpotObjects> beamGThandle;
+
+  auto const& mybeamspot = iSetup.getData(bsToken_);
+  //iSetup.get<BeamSpotTransientObjectsRcd>().get(beamhandle);
+  //const BeamSpotObjects* mybeamspot = beamhandle.product();
+  //iSetup.get<BeamSpotObjectsRcd>().get(beamGThandle);
+  //const BeamSpotObjects* myGTbeamspot = beamGThandle.product();
+
+  edm::LogInfo("Run numver: ") << iEvent.id().run();
+  edm::LogInfo("beamspot from HLT ") << mybeamspot;
+  //edm::LogInfo("beamspot from GT ")<<*myGTbeamspot;
+}
+void OnlineBeamSpotFromDB::fillDescriptions(edm::ConfigurationDescriptions& desc) {
+  edm::ParameterSetDescription dsc;
+  desc.addWithDefaultLabel(dsc);
+}
+
+//define this as a plug-in
+DEFINE_FWK_MODULE(OnlineBeamSpotFromDB);
```

Comment:
If this EDAnalyzer is meant just for tests, how about moving it to `RecoVertex/BeamSpotProducer/test`? You will need a `BuildFile.xml` in the same folder, too.

Reply:
Hi @jpata, I am working on the other comments, about this one, it will be used in  a menu during the next MWGR to check if the payload is correctly loaded, so in principle it is more than for testing locally. This being said, I do not have strong opinion about where it should stay.

Reply:
I see, thanks for the clarification. If this will be used for more than just local tests, then I think it's fine where it is. However, in that case, we should aim to remove the commented-out code. Thanks!

Reply:
Sure, I will remove them.